framework/include/utils/MooseTypes.h

```diff
@@ -26,6 +26,7 @@
 #include "Eigen/Core"
 #include "libmesh/restore_warnings.h"
 #include "libmesh/tensor_tools.h"
+#include "libmesh/dense_matrix.h"
```

Comment:
Instead of including this header, please just use a forward declaration:
```c++
namespace libMesh
{
template <typename>
class DenseMatrix;
}
```
since you are only doing `typedefs` here. You do not need the complete type